torch/quantization/make_module.py

```diff
@@ -0,0 +1,50 @@
+import torch
+
+_operation_from_float = {
+    torch.add: torch.ops.quantized.add,
+    torch.cat: torch.ops.quantized.cat
+}
+
+"""Factory method to create a module wrapper.
+
+Args:
+    op: The operation to wrap the module around
+    quantized: Flag to indicate that the quantized counterpart is required.
+
+Returns:
+    Module class
+
+Example::
+
+    >>> add = make_module(torch.add)
+    >>> q_add = make_module(torch.add, quantized=True)
+"""
+def make_module(op=None, quantized=False):
+    name = op.__name__.capitalize()
+    name = "_ops_" + name
+
+    if quantized:
+        name = name = name + "_quantized"
+        op = _operation_from_float.get(op, op)
+
+    def _init(self, operation):
+        super(type(self), self).__init__()
+        self.operation = operation
+        setattr(torch, name, type(self))  # Add the type to the torch namespace
+
+    def _forward(self, *args, **kwargs):
+        return self.operation(*args, **kwargs)
+
+    def _from_float(self, mod, *args, **kwargs):
+        return make_module(mod.op, name=name, quantized=True)
```

Comment:
Need to add logic to compute and assign scale and zero_point.

Reply:
Suggest creating two functions, one for float and one for quantized: 
```python
def make_module(op):
   name = op.__name__.capitalize()
   name = "_ops_" + name

    def _init(self, operation):
        super(type(self), self).__init__()
        self.operation = operation
        setattr(torch, name, type(self))  # Add the type to the torch namespace

  def _forward(self, *args, **kwargs):
        return self.operation(*args, **kwargs)

    _methods = {
        "__doc__": r"""Wrapper module around function {}.""".format(type(op)),
        "__init__": _init,
        "forward": _forward
    }
   module = type(name, (torch.nn.Module,), _methods)
   return module(op)
```
and 
```
def make_quantized_module(op):
   name = op.__name__.capitalize()
   name = "_ops_" + name+"_quantized"

    def _init(self, operation):
        super(type(self), self).__init__()
        self.operation = operation
        self.scale = 1.0
        self.zero_point = 0
        setattr(torch, name, type(self))  # Add the type to the torch namespace

  def _forward(self, *args, **kwargs):
        return self.operation(*args, **kwargs, scale = self.scale, zp = self.zero_point)

    _methods = {
        "__doc__": r"""Wrapper module around function {}.""".format(type(op)),
        "__init__": _init,
        "forward": _forward
    }
   module = type(name, (torch.nn.Module,), _methods)
   return module(op)
```

Reply:
Should I just generate the default values, or should I extract it from somewhere?

Reply:
Rewrote the whole thing -- now a set of files are generated based on the implemented ops